echoregions/tests/test_regions2d.py

```diff
@@ -0,0 +1,721 @@
+    """
+    Test parsing of Regions2D object via checking individual values and aggregate values.
+
+    Parameters
+    ----------
+    lines_fixture : Lines
+        Object containing data of test EVL file.
+    """
+
+    # Get dataframe
+    df_r2d = regions2d_fixture.data
+
+    # Check shape
+    assert df_r2d.shape == (19, 22)
+
+    # Check individual values of specific row
+    assert df_r2d.loc[4]["file_name"] == "transect.evr"
+    assert df_r2d.loc[4]["file_type"] == "EVRG"
+    assert df_r2d.loc[4]["evr_file_format_number"] == "7"
+    assert df_r2d.loc[4]["echoview_version"] == "13.0.378.44817"
+    assert df_r2d.loc[4]["region_id"] == 5
+    assert df_r2d.loc[4]["region_structure_version"] == "13"
+    assert df_r2d.loc[4]["region_name"] == "ST22"
+    assert df_r2d.loc[4]["region_class"] == "Log"
+    assert df_r2d.loc[4]["region_creation_type"] == "6"
+    assert df_r2d.loc[4]["region_bbox_left"] == pd.to_datetime(
+        "2019-07-02 13:14:00.488000"
+    )
+    assert df_r2d.loc[4]["region_bbox_right"] == pd.to_datetime(
+        "2019-07-02 13:14:01.888000"
+    )
+    assert df_r2d.loc[4]["region_bbox_top"] == -9999.99
+    assert df_r2d.loc[4]["region_bbox_bottom"] == 9999.99
+    assert df_r2d.loc[4]["region_point_count"] == "4"
+    assert df_r2d.loc[4]["region_selected"] == "0"
+    assert df_r2d.loc[4]["dummy"] == "-1"
+    assert df_r2d.loc[4]["region_bbox_calculated"] == 1
+    assert df_r2d.loc[4]["region_type"] == "2"
+    assert (
+        df_r2d.loc[4]["time"]
+        == [
+            np.datetime64("2019-07-02T13:14:00.488000000"),
+            np.datetime64("2019-07-02T13:14:00.488000000"),
+            np.datetime64("2019-07-02T13:14:01.888000000"),
+            np.datetime64("2019-07-02T13:14:01.888000000"),
+        ]
+    ).all()
+    assert (df_r2d.loc[4]["depth"] == [-9999.99, 9999.99, 9999.99, -9999.99]).all()
+    assert df_r2d.loc[4]["region_notes"] == [
+        "Starting transect x22 at 22.0, going ~5 kts to allow a barge to pass"
+    ]
+    assert df_r2d.loc[4]["region_detection_settings"] == []
+
+
+@pytest.mark.regions2d
+def test_evr_to_file(regions2d_fixture: Regions2D) -> None:
+    """
+    Test converting an Echoview 2D Regions files (.EVR).
+    """
+
+    # Get output path
+    output_csv = DATA_DIR / "output_CSV/"
+
+    # Create CSV
+    regions2d_fixture.to_csv(output_csv)
+
+    # Remove files
+    assert os.path.exists(regions2d_fixture.output_file[0])
+    os.remove(regions2d_fixture.output_file[0])
+
+    # Remove directories
+    os.rmdir(output_csv)
+
+
+@pytest.mark.regions2d
+def test_plot(regions2d_fixture: Regions2D) -> None:
+    """
+    Test plotting Regions2D object without errors.
+
+    Parameters
+    ----------
+    regions2d_fixture : Regions2D
+        Object containing data of test EVR file.
+    """
+
+    regions2d_fixture.plot([11], color="k")
+
+
+@pytest.mark.regions2d
+def test_select_sonar_file(regions2d_fixture: Regions2D) -> None:
```

Comment:
maybe test the case when sonar filenames do not follow expected format. Can also add in the documentation of the `select_sonar_file` that is specific to SIMRAD format and write it out how it looks like.

Reply:
Sounds good. Added a test and checks for valid SIMRAD and sonar_file_inputs.